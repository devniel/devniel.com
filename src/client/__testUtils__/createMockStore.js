import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const createReduxMockStore = (defaultState = {}) => {
  const store = mockStore(defaultState);
  return store;
};

export default createReduxMockStore;
