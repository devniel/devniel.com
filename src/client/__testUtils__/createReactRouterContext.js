import { shape } from 'prop-types';
import { BrowserRouter } from 'react-router-dom';

const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
};

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) },
});

export default createContext;
