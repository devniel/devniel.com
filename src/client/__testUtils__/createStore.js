import { createBrowserHistory } from 'history';
import createStore from '../../helpers/createStore';

const createReduxStore = () => {
  const history = createBrowserHistory();
  const store = createStore(history);
  return store;
};

export default createReduxStore;
