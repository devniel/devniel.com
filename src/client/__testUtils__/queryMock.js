import { QueryMock } from '@devniel/graphql-query-test-mock';

export const queryMock = new QueryMock();

export const GRAPHQL_API_URL = 'https://localhost:3003/graphql';
