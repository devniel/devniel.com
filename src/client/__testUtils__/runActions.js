import reducers from '../reducers';

export default function runActions(_actions) {
  let state;
  _actions.forEach(action => {
    const result = reducers(state, action);
    state = result;
  });
  return state;
}
