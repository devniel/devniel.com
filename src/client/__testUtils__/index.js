import wait from './wait';
import runActions from './runActions';
import { queryMock, GRAPHQL_API_URL } from './queryMock';
import createReactRouterContext from './createReactRouterContext';
import createStore from './createStore';
import createMockStore from './createMockStore';
import shallowUntilTarget from './shallowUntilTarget';

export {
  wait,
  runActions,
  queryMock,
  GRAPHQL_API_URL,
  createReactRouterContext,
  createStore,
  createMockStore,
  shallowUntilTarget,
};
