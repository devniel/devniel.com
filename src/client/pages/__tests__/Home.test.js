import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createMockStore } from '../../__testUtils__';
import svg from '../../__testUtils__/fakes/svg';

import sinon from 'sinon';

import HomeConnected from '../Home';

describe('Home HOC', () => {
  let store;
  let root;

  let fakeXHR;
  let requests;
  let dispatchSpy;

  beforeEach(async () => {
    // To attend ReactSVG xhr requests
    fakeXHR = sinon.useFakeXMLHttpRequest();
    requests = [];
    fakeXHR.onCreate = xhr => {
      requests.push(xhr);
    };

    store = createMockStore();

    // It's important to create the spy on store.dispatch before
    // providing it.
    dispatchSpy = sinon.spy(store, 'dispatch');

    root = mount(
      <Provider store={store}>
        <HomeConnected.component />
      </Provider>
    );

    requests.forEach(request => {
      request.respond(200, {}, svg);
    });

    //jest.runAllTimers();
  });

  afterEach(() => {
    fakeXHR.restore();
  });

  it('should render correctly', () => {
    expect(root).toMatchSnapshot();
  });

  it('when clicking the Dail component an action should be queued', async () => {
    const dail = root.find('Dail');
    dail.simulate('click');
    sinon.assert.calledOnce(dispatchSpy);
  }, 500000);
});
