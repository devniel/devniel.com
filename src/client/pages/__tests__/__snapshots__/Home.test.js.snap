// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home HOC should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Home)>
    <Home
      dispatch={[Function]}
    >
      <div
        className="Home"
      >
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
        >
          <SideEffect(NullComponent)
            defer={true}
            encodeSpecialCharacters={true}
            link={
              Array [
                Object {
                  "href": "apple-touch-icon.png",
                  "rel": "apple-touch-icon",
                  "sizes": "180x180",
                },
                Object {
                  "href": "favicon-32x32.png",
                  "rel": "icon",
                  "sizes": "32x32",
                  "type": "image/png",
                },
                Object {
                  "href": "favicon-16x16.png",
                  "rel": "icon",
                  "sizes": "16x16",
                  "type": "image/png",
                },
                Object {
                  "href": "site.webmanifest",
                  "rel": "manifest",
                },
              ]
            }
            meta={
              Array [
                Object {
                  "charset": "utf-8",
                },
              ]
            }
            title="Daniel Flores ~ @devniel, making ideas happen."
            titleAttributes={Object {}}
          >
            <NullComponent
              defer={true}
              encodeSpecialCharacters={true}
              link={
                Array [
                  Object {
                    "href": "apple-touch-icon.png",
                    "rel": "apple-touch-icon",
                    "sizes": "180x180",
                  },
                  Object {
                    "href": "favicon-32x32.png",
                    "rel": "icon",
                    "sizes": "32x32",
                    "type": "image/png",
                  },
                  Object {
                    "href": "favicon-16x16.png",
                    "rel": "icon",
                    "sizes": "16x16",
                    "type": "image/png",
                  },
                  Object {
                    "href": "site.webmanifest",
                    "rel": "manifest",
                  },
                ]
              }
              meta={
                Array [
                  Object {
                    "charset": "utf-8",
                  },
                ]
              }
              title="Daniel Flores ~ @devniel, making ideas happen."
              titleAttributes={Object {}}
            />
          </SideEffect(NullComponent)>
        </HelmetWrapper>
        <div
          className="Home__header"
        >
          <div
            className="Logo"
          >
            <span
              className="Logo__title"
            >
              devniel
            </span>
            <span
              className="Logo__subtitle"
            >
              just hack
            </span>
          </div>
        </div>
        <div
          className="Home__body"
        >
          <div
            className="Logo"
          >
            <span
              className="Logo__title"
            >
              devniel
            </span>
            <span
              className="Logo__subtitle"
            >
              just hack
            </span>
          </div>
          <Dail
            onClick={[Function]}
            status="making ideas happen"
          >
            <div
              className="Dail"
              onClick={[Function]}
              onKeyUp={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="Dail__image"
              >
                <icon.svg />
              </div>
              <div
                className="Dail__status"
              >
                making ideas happen
              </div>
            </div>
          </Dail>
        </div>
        <div
          className="Home__footer"
        >
          <div
            className="Home__footer-left"
          >
            <BrowserView>
              <div>
                A crafter building things with code
                <br />
                and full-stack software engineer @
                <strong>
                  IBM
                </strong>
                ;
                <br />
                from 
                <em>
                  GraphQL
                </em>
                 with 
                <em>
                  TypeScript
                </em>
                 and 
                <em>
                  Node
                </em>
                 development,
                <br />
                to conversational bots with 
                <em>
                  IBM Watson
                </em>
                 on mobile experiences
                <br />
                 built with 
                <em>
                  React Native
                </em>
                 or desktop clients with 
                <em>
                  Electron
                </em>
                .
              </div>
            </BrowserView>
            <MobileView />
          </div>
          <div
            className="Home__footer-right"
          >
            <div
              className="Home__footer-links"
            >
              <div
                className="Home__footer-link"
              >
                <a
                  href="https://github.com/devniel"
                  target="_blank"
                  title="my github"
                >
                  <github.svg />
                </a>
              </div>
              <div
                className="Home__footer-link"
              >
                <a
                  href="Daniel_Flores_Software-Engineer-IBM_Resume_2019_May.pdf"
                  target="_blank"
                  title="my resume"
                >
                  <resume.svg />
                </a>
              </div>
              <div
                className="Home__footer-link"
              >
                <a
                  href="https://twitter.com/devniel"
                  target="_blank"
                  title="my twitter"
                >
                  <twitter.svg />
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Home>
  </Connect(Home)>
</Provider>
`;
